core/xml.js

```diff
@@ -131,6 +133,18 @@ Blockly.Xml.allFieldsToDom_ = function(block, element) {
  * @return {!Element} Tree of XML elements.
  */
 Blockly.Xml.blockToDom = function(block, opt_noId) {
+  // Skip over insertion markers.
+  if (block.isInsertionMarker()) {
+    var child = block.getChildren(false)[0];
+    if (child) {
+      return Blockly.Xml.blockToDom(child);
+    } else {
+      // Disappears when appended.
+      return /** @type{!Element} */ (new DocumentFragment());
```

Comment:
``domToBlockHeadless_`` is only excluding text nodes, what happens when this round-trips?

Reply:
The document fragment kinda seems to disappear when it's appended? I'm not exactly sure what's going on, you'll have to ask @NeilFraser The document fragment doesn't exist if you log the xml, and it doesn't get added to the string when you run domToText.

I added a call to domToWorkspace in the assertXml function. It won't test the structure of the xml (toString should do that) but it will make sure that round-tripping the xml doesn't throw errors.